src/function.py

```diff
@@ -150,21 +150,23 @@ def upload_and_create(client: CogniteClient, config: FunctionConfig) -> Function
 
     if config.overwrite:
         # upsert was requested. delete schedules, function and files
-        try_delete(client=client, external_id=config.external_id)
-
+        try_delete(client, config.external_id)
     try:
         file_id = zip_and_upload_folder(client, config, zip_file_name)
         return create_function_and_wait(client=client, file_id=file_id, config=config)
 
+    except CogniteAPIError as e:
+        if "Function externalId duplicated" in e.message:
```

Comment:
this is a bit dangerous to rely on the error content.

Reply:
Yup, but I have no other ideas if we want to keep our `retry` as is 🙁